server/planning/assignments/assignments.py

```diff
@@ -829,6 +829,20 @@ def is_assignment_being_activated(self, updates, original):
         return original.get('assigned_to').get('state') == ASSIGNMENT_WORKFLOW_STATE.DRAFT and\
             updates.get('assigned_to', {}).get('state') == ASSIGNMENT_WORKFLOW_STATE.ASSIGNED
 
+    def is_text_assignment(self, assignment):
+        text_assignment = False
+        text_types = []
+        content_types = get_resource_service('vocabularies').find_one(req=None, _id='g2_content_type')
+        if content_types:
+            text_types = [t.get('qcode') for t in (content_types.get('items') or [])
+                          if t.get('content item type') == 'text']
+            text_assignment = assignment.get('planning', {}).get('g2_content_type') in text_types
+
+        if len(text_types) == 0 and assignment.get('planning', {}).get('g2_content_type') == 'text':
```

Comment:
Same goes here, simply get the item from `content_types` via `qcode`, and get the `content item type`, defaulting to `qcode`.